integration/helpers/instance.go

```diff
@@ -597,11 +604,26 @@ func (i *TeleInstance) CreateEx(t *testing.T, trustedSecrets []*InstanceSecrets,
 	return i.CreateWithConf(t, tconf)
 }
 
+func (i *TeleInstance) createTeleportProcess(tconf *servicecfg.Config) (*service.TeleportProcess, error) {
+	if i.ProcessProvider == nil {
+		p, err := service.NewTeleport(tconf)
+		if err != nil {
+			return nil, err
+		}
+		return p, nil
+	}
+	p, err := i.ProcessProvider.NewTeleport(tconf)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return p, nil
```

Comment:
Suggestion:

```suggestion
	newTeleport := i.ProcessProvider
	if newTeleport == nil {
		newTeleport = service.NewTeleport
	}

	p, err := newTeleport(tconf)
	return p, trace.Wrap(err)
```

Reply:
This will fail because in this case Go doesn't allow to assign a raw function to object interface. So the following call assignment is invalid `newTeleport = service.NewTeleport`. This will work if the ProcessProvider is just a function but I think that injecting objects with right interface is more readable solution.

Reply:
My bad, that should be:

```go
	newTeleport := service.NewTeleport
	if i.ProcessProvider != nil {
		newTeleport = i.ProcessProvider.NewTeleport
	}
```

It's already merged so don't worry about it.